### PR TITLE
Add UserAssigned identity support to cluster-azure

### PR DESCRIPTION
**Pull request by primeroz**
### Added

- Add support for creating cluster with `UserAssigned Identity` for `VM Identity` 
  - this is optional at the moment and disabled by default since we need the `Identity Operator` to make it work for WC Clusters. It is used only on the MC Cluster when created with https://github.com/giantswarm/cluster-azure/pull/9 and values set to true in LastPass for a new cluster 

### Changed

- replace version with `0.0.0-dev` in Chart.yaml since we use App Build Suite
- Allow customizing the `identityRef` in the `AzureCluster`
- Fix MachinePool naming by removing the hashed name from all resources. This is not needed for MachinePools , like it is for MachineDeployments
- Skip `coredns` installation phase in `kubeadmbootstrapconfiguration` , we install it as an App
- Do not consider the `labels` in the ControlPlane AzureMachineTemplate when calculating name hash to avoid rolling control plane nodes unecessarily
- Change default values ssh key to RSA one ( since azure does not support ed25519 )
- Update schema json

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+repos:
```
Comment (bavarianbidi) [inline .pre-commit-config.yaml:1]: i like the idea of the `pre-commit` configuration but i would propose:

1. move this into a separate PR
2. ask in #sig-dev/#sig-releng about opinions (maybe it's worth to enable globally via our automation)

**Review by bavarianbidi — COMMENTED** [review]
LGTM

File: .pre-commit-config.yaml (hunk 1 above)
Comment (primeroz) [reply .pre-commit-config.yaml]: removed.

**Review by bavarianbidi — APPROVED** [review]
